nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -246,14 +266,15 @@ BtNavigator::navigateToPose()
       return action_server_->is_cancel_requested();
     };
 
-  auto bt_xml_filename = action_server_->get_current_goal()->behavior_tree;
+  std::string bt_xml_filename = action_server_->get_current_goal()->behavior_tree.c_str();
```

Comment:
I don't think `c_str()` is necessary. You're casting a string into a char* back into a string

Reply:
It is not needed. That is correct. I'll delete it (just tested.. remembered there were some issues. Probably only during "printing")